sql/src/main/java/io/crate/memory/OnHeapMemoryManager.java

```diff
@@ -38,6 +38,8 @@ public OnHeapMemoryManager(IntConsumer accountBytes) {
     @Override
     public ByteBuf allocate(int capacity) {
         accountBytes.accept(capacity);
+        // TODO: should we hold onto the buffers and release them as well?
+        // contract would require it, but there is no resource behind it that must be released
         return Unpooled.wrappedBuffer(new byte[capacity]);
```

Comment:
any reasons not to use platform aware `return Unpooled.buffer(capacity)` ?

Reply:
For some reason I thought that `buffer` could be either heap or direct memory - depending on netty system properties. But you're right it is always doing heap allocations - I'll change it.